pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -118,10 +136,57 @@ func (cache *claimInfoCache) hasPodReference(UID types.UID) bool {
 	defer cache.RUnlock()
 
 	for _, claimInfo := range cache.claimInfo {
-		if claimInfo.podUIDs.Has(string(UID)) {
+		if claimInfo.PodUIDs.Has(string(UID)) {
 			return true
 		}
 	}
 
 	return false
 }
+
+func (cache *claimInfoCache) addPodReference(claimName, namespace string, podUID types.UID) error {
+	cache.Lock()
+	defer cache.Unlock()
+
+	claimInfo, exist := cache.claimInfo[claimName+namespace]
+	if !exist {
+		return fmt.Errorf("failed to get claimInfo with claim name %s and namespace %s", claimName, namespace)
+	}
+
+	claimInfo.PodUIDs.Insert(string(podUID))
+	return nil
+
+}
+
+func (cache *claimInfoCache) deletePodReference(claimName, namespace string, podUID types.UID) error {
+	cache.Lock()
+	defer cache.Unlock()
+
+	claimInfo, exist := cache.claimInfo[claimName+namespace]
+	if !exist {
+		return fmt.Errorf("failed to get claimInfo with claim name %s and namespace %s", claimName, namespace)
+	}
+
+	claimInfo.PodUIDs.Delete(string(podUID))
+	return nil
+}
+
+func (cache *claimInfoCache) syncToCheckpoint() error {
+	cache.RLock()
+
+	claimInfoStateList := make(state.ClaimInfoStateList, 0)
+	for _, infoClaim := range cache.claimInfo {
+		claimInfoState := state.ClaimInfoState{
+			DriverName: infoClaim.DriverName,
+			ClaimUID:   infoClaim.ClaimUID,
+			ClaimName:  infoClaim.ClaimName,
+			Namespace:  infoClaim.Namespace,
+			PodUIDs:    infoClaim.PodUIDs,
+			CdiDevices: infoClaim.CdiDevices,
+		}
+		claimInfoStateList = append(claimInfoStateList, claimInfoState)
+	}
+	cache.RUnlock()
```

Comment:
Given my comment above it may be better to just not make a copy and instead hold the read lock across the store call. It's easier to reason about and likely a premature optimization not to do it this way.

Reply:
Ok I will update the code to hold the read lock for the store call